tests/components/vizio/test_config_flow.py

```diff
@@ -217,12 +239,34 @@ def vizio_cant_connect_fixture():
     assert result["errors"] == {CONF_NAME: "name_exists"}
 
 
+async def test_user_esn_already_exists(
+    hass: HomeAssistantType, vizio_connect, vizio_bypass_setup
+) -> None:
+    """Test ESN is already configured with different host and name during user setup."""
+    # Set up new entry
+    MockConfigEntry(
+        domain=DOMAIN, data=MOCK_SPEAKER_CONFIG, unique_id=UNIQUE_ID
+    ).add_to_hass(hass)
+
+    # Set up new entry with same unique_id but different host and name
+    fail_entry = MOCK_SPEAKER_CONFIG.copy()
+    fail_entry[CONF_HOST] = HOST2
+    fail_entry[CONF_NAME] = NAME2
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=fail_entry
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "already_setup_with_diff_host_and_name"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4847616)